src/compile_node.rs

```diff
@@ -0,0 +1,114 @@
+use crate::config::FnmConfig;
+use crate::directory_portal::DirectoryPortal;
+use crate::version::Version;
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Can't spawn `node-build`. Do you have it installed? {source}")]
+    SpawningNodeBuild {
+        #[source]
+        source: std::io::Error,
+    },
+    #[error("Can't access `node-build` stdout. Please retry.")]
+    NodeBuildStdoutIsInaccessible,
+    #[error(
+        "`node-build` has failed.\n\n   stdout:\n{}\n\n   stderr: \n{}",
+        stdout,
+        stderr
+    )]
+    ExecutingNodeBuild { stdout: String, stderr: String },
+    #[error("Can't move built version to {target}. {source}")]
+    MovingBuiltVersion {
+        target: std::path::PathBuf,
+        #[source]
+        source: std::io::Error,
+    },
+}
+
+pub fn compile_node_with_node_build(config: &FnmConfig, version: &Version) -> Result<(), Error> {
+    let path = {
+        let mut path = version.installation_path(config);
+        path.pop();
+        path
+    };
+    let temp_installations_dir = tempfile::tempdir().unwrap();
```

Comment:
node-build does not support spaces in directories too. We need to find a solution for that 🥲 
related: 
* https://github.com/Schniz/fnm/issues/622#issuecomment-1029320579

Reply:
I mean, the solution is to have a `dirs` module (which we already have) and probably store stuff in `.local/fnm` or something unless XDG_ env vars are provided, but I want to make the migration fast 🥲